tests/post/features/volume.feature

```diff
@@ -8,3 +8,194 @@ Feature: Volume management
+        Given a Volume 'volume8' exist
+        And the following PersistentVolumeClaim 'my-pvc' exist:
+            kind: PersistentVolumeClaim
+            apiVersion: v1
+            metadata:
+              name: my-pvc
+            spec:
+              storageClassName: metalk8s-prometheus
+              accessModes:
+                  - ReadWriteOnce
+              resources:
+                  requests:
+                    storage: 10Gi
+        And the following Pod 'pod-test-volume-protection' exist:
+            apiVersion: v1
+            kind: Pod
+            metadata:
+              name: pod-test-volume-protection
+            spec:
+              volumes:
+                - name: pod-storage
+                  persistentVolumeClaim:
+                    claimName: my-pvc
+              containers:
+                - name: container-test-volume-protection
+                  image:  busybox
+                  command:
+                  - sleep
+                  - "60"
+                  volumeMounts:
+                    - mountPath: "/mnt/"
+                      name: pod-storage
+              tolerations:
+              - key: "node-role.kubernetes.io/bootstrap"
+                operator: "Exists"
+                effect: "NoSchedule"
+              - key: "node-role.kubernetes.io/infra"
+                operator: "Exists"
+                effect: "NoSchedule"
+              - key: "node-role.kubernetes.io/master"
+                operator: "Exists"
+                effect: "NoSchedule"
+        When I delete the Volume 'volume8'
+        Then the Volume 'volume8' is marked for deletion
+        And the PersistentVolume 'volume8' is marked for deletion
+        And the Volume 'volume8' is 'Available'
+        When I delete the PersistentVolumeClaim 'my-pvc'
+        Then the Volume 'volume8' does not exist
+        And the PersistentVolume 'volume8' does not exist
+
+    Scenario: Complete E2E volume workflow
+        Given the following PersistentVolumeClaim 'e2e-pvc' exist:
+            kind: PersistentVolumeClaim
+            apiVersion: v1
+            metadata:
+              name: e2e-pvc
+            spec:
+              storageClassName: metalk8s-prometheus
+              accessModes:
+                  - ReadWriteOnce
+              resources:
+                  requests:
+                    storage: 10Gi
+        And the following Pod 'pod-test-volume-e2e' exist:
+            apiVersion: v1
+            kind: Pod
+            metadata:
+              name: pod-test-volume-e2e
+            spec:
+              volumes:
+                - name: pod-storage
+                  persistentVolumeClaim:
+                    claimName: e2e-pvc
+              containers:
+                - name: container-test-volume-e2e
+                  image:  busybox
+                  command:
+                  - sh
+                  args: ["-c", 'echo "toto" >> /mnt/toto.log && sleep 60']
```

Comment:
Might be useful here to add a longer sleep interval since the container might exit after the sleep duration.
Also we have to account for the following
- pod creation time
- time for pod to become available(PV and PVC bound time)

I have noticed situations in which the check for pod availability fails.
And by the way, extending the sleep duration to something like 120s is not harmful since we delete this container thereafter.